src/confdb/confdb.c

```diff
@@ -941,17 +941,7 @@ static int confdb_get_domain_internal(struct confdb_ctx *cdb,
     if (tmp == NULL || *tmp == '\0') {
         tmp = "false";
     }
-
-    if (strcasecmp(tmp, "FALSE") == 0) {
-        domain->mpg_mode = MPG_DISABLED;
-    } else if (strcasecmp(tmp, "TRUE") == 0) {
-        domain->mpg_mode = MPG_ENABLED;
-    } else {
-        DEBUG(SSSDBG_FATAL_FAILURE,
-              "Invalid value for %s\n", CONFDB_DOMAIN_AUTO_UPG);
-        ret = EINVAL;
-        goto done;
-    }
+    domain->mpg_mode = str_to_domain_mpg_mode(tmp);
```

Comment:
Here we change the behaviour of the code in the sense that an uknown value gets converted to MPG_DISABLED. I felt like I should point this out in the review.